packages/yourloops/components/dialogs/remove-direct-share-dialog.hook.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2022, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { useTranslation } from 'react-i18next'
+import { useAlert } from '../utils/snackbar'
+import { INotification } from '../../lib/notifications/models'
+import DirectShareApi from '../../lib/share/direct-share-api'
+import { useAuth } from '../../lib/auth'
+import { useNotification } from '../../lib/notifications/hook'
+import { RemoveDirectShareProps } from './remove-direct-share-dialog'
+
+interface RemoveDirectShareDialogHookReturn {
+  isCurrentUserCaregiver: boolean
+  removeDirectShare: () => Promise<void>
+}
+
+const useRemoveDirectShareDialog = ({ userToRemove, onClose }: RemoveDirectShareProps): RemoveDirectShareDialogHookReturn => {
```

Comment:
Looking at the unit tests of this hook made me realize something. `userToRemove` is only used for the method `removeDirectShare`. Instead of passing it to the entire hook, it would be better that it is simply a parameter for the method.

I am also wondering wether `currentUser` should not also be a parameter of this `removeDirectShare`. I'm less sure on that one, what do you think?